### ISSUE TITLE
Examples: how to test services that require auth when the auth itself is not what you're looking to test

### ISSUE THREAD

**Issue by simskij** [grafana/k6-docs#339]
[This forum post](https://community.k6.io/t/setup-function-duration/37) explains the problem:

> I need to make performance test with e.g. 100 users, adding according to stages. All these users must authenticate first. I’ve moved authentication of users into setup() function in purpose to not impact on test execution.
>
> But now it takes at about 60sec to login all users before test starts. And what if I want to simulate 1000 users? Several minutes of waiting?

Mihail [offers some suggestions](https://community.k6.io/t/setup-function-duration/37/2) on how to work with this, and I'm sure there might be additional ones we can come up with.

This could likely be combined with the question in https://community.k6.io/t/how-to-execute-task-only-once/73, "How to execute task only once per VU"

**Comment by immavalls** [comment]
Closing this in favor of https://github.com/grafana/k6-docs/issues/990
